app/client/src/pages/Applications/index.tsx

```diff
@@ -859,36 +861,56 @@ type ApplicationProps = {
   currentUser?: User;
   searchKeyword: string | undefined;
 };
+
+const getIsFromSignup = () => {
+  if (window.location.href) {
+    const url = new URL(window.location.href);
+    const searchParams = url.searchParams;
+    return !!searchParams.get("isFromSignup");
+  }
+  return false;
+};
+const { onboardingFormEnabled } = getAppsmithConfigs();
 class Applications extends Component<
   ApplicationProps,
-  { selectedOrgId: string }
+  { selectedOrgId: string; showOnboardingForm: boolean }
 > {
   constructor(props: ApplicationProps) {
     super(props);
 
     this.state = {
       selectedOrgId: "",
+      showOnboardingForm: false,
     };
   }
 
   componentDidMount() {
     PerformanceTracker.stopTracking(PerformanceTransactionName.LOGIN_CLICK);
     PerformanceTracker.stopTracking(PerformanceTransactionName.SIGN_UP);
     this.props.getAllApplication();
+    this.setState({
+      showOnboardingForm: getIsFromSignup() && onboardingFormEnabled,
+    });
   }
 
   public render() {
     return (
       <PageWrapper displayName="Applications">
-        <ProductUpdatesModal />
-        <LeftPane />
-        <SubHeader
-          search={{
-            placeholder: "Search for apps...",
-            queryFn: this.props.searchApplications,
-          }}
-        />
-        <ApplicationsSection searchKeyword={this.props.searchKeyword} />
+        {!this.state.showOnboardingForm ? (
+          <>
+            <ProductUpdatesModal />
+            <LeftPane />
+            <SubHeader
+              search={{
+                placeholder: "Search for apps...",
+                queryFn: this.props.searchApplications,
+              }}
+            />
+            <ApplicationsSection searchKeyword={this.props.searchKeyword} />
+          </>
+        ) : (
+          <OnboardingForm />
+        )}
```

Comment:
I think this looks cleaner but you can ignore it if you like

```suggestion
        {this.state.showOnboardingForm ? (
           <OnboardingForm />
        ) : (
         <>
            <ProductUpdatesModal />
            <LeftPane />
            <SubHeader
              search={{
                placeholder: "Search for apps...",
                queryFn: this.props.searchApplications,
              }}
            />
            <ApplicationsSection searchKeyword={this.props.searchKeyword} />
          </>
        )}
```

Reply:
yeah i'll update it does look cleaner